tests/MongoDB.Driver.Tests/Specifications/retryable-writes/prose-tests/CommandConstructionTests.cs

```diff
@@ -0,0 +1,332 @@
+                        break;
+
+                    default:
+                        throw new Exception($"Unexpected operation: {operation}.");
+                }
+
+                AssertCommandDoesNotHaveTransactionId(eventCapturer);
+            }
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Unsupported_multi_statement_writes_should_not_have_transaction_id(
+            [Values("deleteMany", "updateMany")] string operation)
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded);
+
+            DropCollection();
+            var eventCapturer = CreateEventCapturer();
+            using (var client = CreateDisposableClient(eventCapturer))
+            {
+                var database = client.GetDatabase(_databaseName);
+                var collection = database.GetCollection<BsonDocument>(_collectionName);
+
+                WriteModel<BsonDocument>[] requests;
+                switch (operation)
+                {
+                    case "deleteMany":
+                        var deleteManyFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        requests = new[] { new DeleteManyModel<BsonDocument>(deleteManyFilter) };
+                        break;
+
+                    case "updateMany":
+                        var updateManyFilter = Builders<BsonDocument>.Filter.Eq("_id", 1);
+                        var update = Builders<BsonDocument>.Update.Set("x", 1);
+                        requests = new[] { new UpdateManyModel<BsonDocument>(updateManyFilter, update) };
+                        break;
+
+                    default:
+                        throw new Exception($"Unexpected operation: {operation}.");
+                }
+                collection.BulkWrite(requests);
+
+                AssertCommandDoesNotHaveTransactionId(eventCapturer);
+            }
+        }
+
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void Aggregate_with_write_stage_should_not_have_transaction_id(
+            [Values("$out", "$merge")] string outStage)
+        {
+            RequireServer.Check().ClusterTypes(ClusterType.ReplicaSet, ClusterType.Sharded);
+
+            DropCollection();
+            var eventCapturer = CreateEventCapturer();
+            using (var client = CreateDisposableClient(eventCapturer))
+            {
+                var database = client.GetDatabase(_databaseName);
+                var collection = database.GetCollection<BsonDocument>(_collectionName);
+
+                PipelineDefinition<BsonDocument, BsonDocument> pipeline = new EmptyPipelineDefinition<BsonDocument>();
+                var outputCollection = database.GetCollection<BsonDocument>(_collectionName + "-outputCollection");
+                switch (outStage)
+                {
+                    case "$out":
+                        pipeline = pipeline.Out(outputCollection);
+                        break;
+
```

Comment:
Minor formatting, do we need to have these blank lines (AFAIK mostly we don't have them)?

Reply:
We usually do because each case is logically distinct from the others.

I'm sure you can find plenty of examples with and without...